whisper/whisperv6/peer.go

```diff
@@ -79,11 +79,14 @@ func (peer *Peer) stop() {
 func (peer *Peer) handshake() error {
 	// Send the handshake status message asynchronously
 	errc := make(chan error, 1)
+	isLightNode := peer.host.LightClientMode()
+	isRestrictedLightNodeConnection := peer.host.LightClientModeConnectionRestricted()
 	go func() {
 		pow := peer.host.MinPow()
 		powConverted := math.Float64bits(pow)
 		bloom := peer.host.BloomFilter()
-		errc <- p2p.SendItems(peer.ws, statusCode, ProtocolVersion, powConverted, bloom)
+
+		errc <- p2p.SendItems(peer.ws, statusCode, ProtocolVersion, powConverted, bloom, isLightNode)
```

Comment:
I'm thinking that our policy should also be known to the client: if we won't connect to light clients, the people who try to connect to us should know.
That's easier for debugging, on the other hand there is the problem of letting peers know that we won't connect if they claim to be light nodes. In that case, they might be tempted to lie. I'm still on the fence about this one.

Reply:
I think this feature is more useful for light peers. It doesn't need to keep an unuseful connection. 
Why could a light peer want to connect to another light peer? They can't receive envelopes from this connection. 
What do you think about changing isLightNode to isLightPeerRestricted? But I think they could lie too.